src/lib.rs

```diff
@@ -412,6 +420,50 @@ fn run_for_context<TContext: Sync>(
     })
 }
 
+pub fn run_for_slice_with_callback(
```

Comment:
This entry point is needed for the use case where `tree-sitter-lint` itself has an in-memory "slice" that it wants to lint (specifically it's using that for its rule-testing helpers)